### PR TITLE
Add support for yum to the Azure DevTest Labs artifact system.

### PR DESCRIPTION
**Pull request by d3r3kk**
# ALM Rangers Artifact: Apt-Get #
## Package Management for Debian Based Systems ##

Exposes the natural package management system for Red Hat-based systems to the Azure DevTest Labs users.

Now users can install whatever software they wish via yum in a way that is natural to people familiar with Red Hat and its derivative distributions (CentOS).

Please see README.md for list of supported distributions and usage.

> NOTE: There is an issue when using the Yum artifact with many of the CentOS images when update is selected. An issue will be pushed shortly to highlight this to the Azure DevTest Labs team. (EDIT: Issue has been raised as #75 )

### CODE REVIEW

File: Artifacts/linux-yum-package/Artifactfile.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        "Linux"
+    ],
+    "iconUri": "http://fedoraproject.org/w/uploads/7/77/Echo-package-48px.png",
+    "targetOsType": "Linux",
+    "parameters": {
+        "packages": {
+            "type": "string",
+            "displayName": "Packages",
+            "allowEmpty": false,
+            "description": "The packages to install on the VM, separated by spaces. All of the package declarations supported by yum are supported here as well, see http://man7.org/linux/man-pages/man8/yum.8.html for an in-depth look at the package specifications available."
```
Comment (drwill-ms) [inline Artifacts/linux-yum-package/Artifactfile.json]: Please spell YUM consistently. It is an acronym so perhaps always in uppercase.

File: Artifacts/linux-yum-package/Artifactfile.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        },
+        "update": {
+            "type": "string",
+            "displayName": "Run Update First",
+            "defaultValue": "true",
+            "description": "Run yum update prior to running the full install command. Allowed values are 'true' and 'false', or blank (which will evaluate to false). Note that no additional options are passed to the update command."
```
Comment (drwill-ms) [inline Artifacts/linux-yum-package/Artifactfile.json]: You can use the allowedValues json property to specify only true/false are allowed

File: Artifacts/linux-yum-package/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+> limitations where double-spaces will be reduced to a single space, and other edge cases that might arise due to its use. If there is a 
+> specific use case that doesn't work within this limitation, there is the linux-bash DevTest artifact that will satisfy your needs.
+
+## Notes on yum
+Note that yum as a utility is an incredibly flexible and powerful tool. To support the use cases of the more advanced users of
+our Linux DevTest Lab VMs, a reasonable effort has been made to support the nuance of the 'install' command. However, due to the vast number of 
+permutations and combinations that the command can be used to manipulate the state of a machine, we recommend that for the most advanced
+of users with specific needs use the linux-bash artifact to satisfy their specific needs.
+
+For posterity, here is the yum man page section on the install command at the time of authoring this artifact:
```
Comment (drwill-ms) [inline Artifacts/linux-yum-package/README.md]: Suggest not to include this. Most people won't be looking at the scripts, they can read the up-to-date man page for themselves, and adding to the this script simply increases the file size for deploy/storage unnecessarily.

**Comment by d3r3kk** [comment]
This PR is also updated. Is there anything else required?

**Pull request merged**